ts/types/Attachment.ts

```diff
@@ -41,6 +40,26 @@ export const isVisualMedia = (attachment: Attachment): boolean => {
   return isSupportedImageType || isSupportedVideoType;
 };
 
+export const isVoiceMessage = (attachment: Attachment): boolean => {
+  const flag = SignalService.AttachmentPointer.Flags.VOICE_MESSAGE;
+  const hasFlag =
+    // tslint:disable-next-line no-bitwise
+    !is.undefined(attachment.flags) && (attachment.flags & flag) === flag;
+  if (hasFlag) {
+    return true;
+  }
+
+  const isLegacyAndroidVoiceMessage =
+    !is.undefined(attachment.contentType) &&
+    MIME.isAudio(attachment.contentType) &&
+    attachment.fileName === null;
+  if (isLegacyAndroidVoiceMessage) {
+    return true;
+  }
```

Comment:
This check for legacy voice messages required `fileName: string | null | undefined` type.